packages/data-stores/src/site/actions.ts

```diff
@@ -95,18 +95,24 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		type: 'RESET_RECEIVE_NEW_SITE_FAILED' as const,
 	} );
 
-	const launchedSite = ( siteId: number ) => ( {
-		type: 'LAUNCHED_SITE' as const,
+	const launchSiteStart = ( siteId: number ) => ( {
+		type: 'LAUNCH_SITE_START' as const,
+		siteId,
+	} );
+
+	const launchSiteComplete = ( siteId: number ) => ( {
+		type: 'LAUNCH_SITE_COMPLETE' as const,
 		siteId,
 	} );
 
 	function* launchSite( siteId: number ) {
+		yield launchSiteStart( siteId );
 		yield wpcomRequest( {
 			path: `/sites/${ siteId }/launch`,
 			apiVersion: '1.1',
 			method: 'post',
 		} );
-		yield launchedSite( siteId );
+		yield launchSiteComplete( siteId );
```

Comment:
We should add error handling on the `yield wpcomRequest()` call. Even an empty `try / catch` is probably better than having this action failing mid-way through.

Ideally, though, we should add a new `LAUNCH_SITE_ERROR` action, and handle that property in the reducer/selectors too. And therefore, also have the "launch error" UI ready. Not a blocker for this PR, but should probably be added to the backlog

Reply:
Yes, good point. LAUNCH_SITE_ERROR action would also remove the loading state.

Reply:
Yes, that's what I was thinking too. We should also add a flag in the reducer for when there is an error, and probably add a simple error message in the "Success" view, similar to what you already added for the loading state

Reply:
Added https://github.com/Automattic/wp-calypso/issues/47782 to handle this case and check the rest of the flow.